### PR TITLE
Implemented static category links

### LINKED ISSUE #128: Proposed Feature: Static Category Link Buttons (Remix)
There have been requests for the return of the static category link buttons in the old Nyaa (as in #106 and even in #92 ). I am proposing the following redesign that would be both modern and colorful :)

> 
> 
> 

Such design can also include a more complicated control flow. A small open button, which will reveal the subcategories upon clicks, can be added to the end each parent category link button. Though it may slightly duplicate the function provided by the search bar. 

P.S. The background colors are for reference only.

### PR DESCRIPTION
**Pull request by UnKnoWn-Consortium**
Re-make of #162 
Fixed #128 

Sorry that I have to close the old request and make a new one. Next time I will use rebase :)

### CODE REVIEW

**Comment by shm0o** [comment]
Could you also squash that last commit?

**Comment by sharkykh** [comment]
Don't forget to bump the CSS version [here](https://github.com/UnKnoWn-Consortium/nyaa/blob/4edce8aa1288c6ceb6081b405c45e3e13aba4c01/nyaa/templates/layout.html#L31).
Compared to https://github.com/nyaadevs/nyaa/blob/master/nyaa/templates/layout.html#L31

**Comment by UnKnoWn-Consortium** [comment]
@sharkykh I am wondering whether I need to bump it one more version up (i.e. v=13 instead of v=12) if my changed files include the main.css?

**Comment by sharkykh** [comment]
@UnKnoWn-Consortium Yes, exactly. Every time you want to push (commit to nyaa.si) an updated `main.css` or `main.js` you need to bump the version by 1 from the current version set on master branch.

**Comment by TheAMM** [comment]
Actually @UnKnoWn-Consortium, @sharkykh, you may get rid of the `?v=` (and prepare for fixing merge conflicts) - I've pushed an automated cachebuster helper in 1ab3d6e3eb1cc521e4f5c4ddd05a7c72f9d6bc5a

**Comment by sharkykh** [comment]
@TheAMM Wonderful, thanks!

**Comment by aldacron** [comment]
wait, what is the screenshot result for this?

**Comment by UnKnoWn-Consortium** [comment]
@aldacron 
Oops! I have forgotten to move back the thumbnails from the closed pull requests





Scrollable when the screen width is insufficient

**Comment by UnKnoWn-Consortium** [comment]
Conflicts have been resolved and the pull request itself has been tested in the current master :)

File: nyaa/templates/layout.html [hunk 1]
```diff
@@ -293,6 +436,35 @@
...
+								{% if search is defined and search.category != '0_0' and search.category[0] != main_cat['id'] %} hidden {% endif %}
+								{% if search is defined and search.category[0] == main_cat['id'] %} active {% endif %}
+								{% if search is defined and search.category[-1] == '0' %} last {% endif %}">
+						<a href="/{% if search is defined and search.user is not none %}user/{{ user.username }}{% endif %}?q=&f=0&c={{ main_cat['id'] }}_0">{{ main_cat['name'] }}</a>
+					</li>
+						{% for sub_cat in main_cat["sub_cats"] %}
+						<li class="{% if sub_cat['name'] == sub_cat['title'] %} main-cat {% else %} sub-cat {% endif %}
+									{% if search is defined and search.category[-1] == sub_cat['id'] %} active last {% endif %}
+									{% if search is defined and search.category[0] != main_cat['id'] %} hidden {% endif %}">
+							<a href="/{% if search is defined and search.user is not none %}user/{{ user.username }}{% endif %}?q=&f=0&c={{ main_cat['id'] }}_{{ sub_cat['id'] }}">{{ sub_cat['name'] }}</a>
```
Comment (shm0o) [inline nyaa/templates/layout.html]: Please, build these URLs with Flask's `url_for`.

**Comment by aldacron** [comment]
a few things:

1) when you put the site in Sukebei, not all the categories show -- only the top level categories show
2) I love the idea of static links and I think that would help discoverability, but I talked to some people about it and I don't think we're sold on having another "bar" since that takes up precious real estate. I think a good compromise here is to have another dropdown menu in the top bar:

-  the top level menu will be the top level categories
- the submenus will be the sub categories. 

does this make sense?
3) finally, just a heads up and when we merge #121 this _will_ need a pretty big refactor

**Comment by Arylide** [comment]
https://github.com/nyaadevs/nyaa/pull/391#issuecomment-376015081

**Pull request closed without merging**